inception-recommendation-api/src/main/resources/META-INF/asciidoc/developer-guide/implementing_recommender_tutorial.adoc

```diff
@@ -0,0 +1,362 @@
+// Copyright 2019
+// Ubiquitous Knowledge Processing (UKP) Lab
+// Technische Universität Darmstadt
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[sect_recommender_tutorial]]
+= Implementing a custom recommender
+
+This section describes the overall design of internal recommenders in INCEpTION and gives a
+tutorial on how to implement them. Internal recommenders are directly added to the INCEpTION platform
+by implementing Java interfaces. These are then picked up during load time by the Spring Framework.
+
+For this tutorial, we will add a recommender for named entities that uses the data majority label for
+predicting, i.e. it predicts always the label that appears most often in the training data. The full
+code for this example can be found on https://github.com/inception-project/inception/inception-imls-data-majority[on Github].
```

Comment:
Link does not work and file doesn't seem to be included in PR - also doesn't seem to be tested.